hoomd/data/typeparam.py

```diff
@@ -1,14 +1,17 @@
 from hoomd.data.parameterdicts import AttachedTypeParameterDict
-from copy import deepcopy
 
 
 class TypeParameter:
+    __slots__ = ('name', 'type_kind', 'param_dict')
```

Comment:
I can't recall the details, but there are a lot of potential problems with `__slots__` if class inheritance is involved. I believe that using `__slots__` for filtering class attributes in `__getattr__` wasn't an option in signac's synced collections for that reason. There are some notes here: https://docs.python.org/3/reference/datamodel.html#notes-on-using-slots

Reply:
Yes, for classes that will likely be inherited slots are not a good option. This is not the case here though.